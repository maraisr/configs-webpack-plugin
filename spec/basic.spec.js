const rimraf = require('rimraf');
const { expectStatsGreen, runWebpack } = require('./helpers');
const { join } = require('path');
const { WebpackRuntimeConfig } = require('../');

const basicConfig = {
	somethingValue: 'fooBar',
	shouldTreeShake: 'tree-shaken-value',
	nested: {
		test: 'nested',
	},
};

const OUTPUT_DIR = join(__dirname, 'fixtures/basic-dist/');

describe('WebpackRuntimeConfig :: basic', () => {
	beforeEach(done => {
		rimraf(OUTPUT_DIR, done);
	});

	it('should build with async', async () => {
		const stats = await runWebpack({
			plugins: [new WebpackRuntimeConfig({
				configs: [
					{ name: 'dev', config: basicConfig },
					{ name: 'uat', config: { ...basicConfig, somethingElseHappened: 'in uat' } },
				],
				request: 'gdu/config',
			})],
		}, OUTPUT_DIR);

		expectStatsGreen(stats);
	});

	it('should fire hook for configChunks', async () => {
		// one for the webpack errors, and one for the hook.
		expect.assertions(2);

		await runWebpack({
			plugins: [
				new class TesterPlugin {
					apply(compiler) {
						compiler.hooks.compilation.tap('testing', compilation => {
							WebpackRuntimeConfig.getHooks(compilation)
								.configChunks.tap('tester', configChunks => {
								expect(configChunks).toHaveLength(2);
							});
						});
					}
				},
				new WebpackRuntimeConfig({
					configs: [
						{ name: 'dev', config: basicConfig },
						{ name: 'uat', config: { ...basicConfig, somethingElseHappened: 'in uat' } },
					],
					request: 'gdu/config',
				}),
			],
		}, OUTPUT_DIR);
	});

	it.each([
		'./index.js',
		'./defered.js',
	])('should match snapshot for %s', async (entry) => {
		const stats = await runWebpack({
			entry,
			plugins: [
				new class TestingPlugin {
					apply(compiler) {
						compiler.hooks.afterCompile.tap('testing', compilation => {
							Object.entries(compilation.assets)
								.forEach(([name, asset]) => {
									expect(asset.source())
										.toMatchSnapshot(name);
								});
						});
					}
				},
				new WebpackRuntimeConfig({
					runtimePublicPath: 'publicPath',
					configs: [
						{ name: 'dev', config: basicConfig },
						{ name: 'uat', config: { ...basicConfig, somethingElse: 'in uat' } },
						{ name: 'pre', config: { ...basicConfig, publicPath: 'public_path_value' } },
					],
					request: 'gdu/config',
				}),
			],
		}, OUTPUT_DIR);

		expectStatsGreen(stats);
	});

	it('should match snapshot', async () => {
		await runWebpack({
			plugins: [
				new class TestingPlugin {
					apply(compiler) {
						let chunks;

						compiler.hooks.compilation.tap('testing', compilation => {
							WebpackRuntimeConfig.getHooks(compilation)
								.configChunks
								.tap('testing', configChunks => {
									chunks = configChunks;
								});
						});

						compiler.hooks.afterCompile.tap('testing', compilation => {
							const chunkFiles = chunks.map(item => item.files).flat();

							for (const configChunkFileName of chunkFiles) {
								expect(compilation.assets[configChunkFileName]).toBeTruthy();

								expect(compilation.assets[configChunkFileName].source())
									.toMatchSnapshot(configChunkFileName);
							}
						});
					}
				},
				new WebpackRuntimeConfig({
					runtimePublicPath: 'publicPath',
					configs: [
						{ name: 'dev', config: basicConfig },
						{ name: 'uat', config: { ...basicConfig, somethingElse: 'in uat' } },
						{ name: 'pre', config: { ...basicConfig, publicPath: 'public_path_value' } },
					],
					request: 'gdu/config',
				}),
			],
		}, OUTPUT_DIR);
	});

	it('should work in production mode also', async () => {
		const stats = await runWebpack({
			mode: 'production',
			optimization: {
				concatenateModules: true,
			},
			plugins: [new WebpackRuntimeConfig({
				configs: [
					{ name: 'dev', config: basicConfig },
				],
				request: 'gdu/config',
			})],
		}, OUTPUT_DIR);

		expectStatsGreen(stats);
	});

	it('should allow runtime public path', async () => {
		const stats = await runWebpack({
			plugins: [
				new WebpackRuntimeConfig({
					runtimePublicPath: 'publicPathSettingPath',
					configs: [
						{
							name: 'dev',
							config: {
								...basicConfig,
								publicPathSettingPath: 'publicPathSettingValue',
							},
						},
						{
							name: 'uat',
							config: {
								...basicConfig,
							},
						},
					],
					request: 'gdu/config',
				}),
			],
		}, OUTPUT_DIR);

		expectStatsGreen(stats);
	});
});
