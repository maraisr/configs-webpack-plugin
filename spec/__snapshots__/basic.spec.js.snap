// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./defered.js: 0.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/*!********************!*\\\\
  !*** config query ***!
  \\\\********************/
/*! no static exports found */
/*! all exports used */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony import */ var __CONFIG__ = __webpack_require__(/*! config_module */ \\"config_module\\");
var __CONFIG___namespace = /*#__PURE__*/__webpack_require__.t(/*! config_module */ \\"config_module\\", 1);
/* harmony reexport (unknown) */ for(var __WEBPACK_IMPORT_KEY__ in __CONFIG__) if(__WEBPACK_IMPORT_KEY__ !== 'default') (function(key) { __webpack_require__.d(exports, key, function() { return __CONFIG__[key]; }) }(__WEBPACK_IMPORT_KEY__));


/***/ })
]]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./defered.js: config-dev.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-dev\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for dev ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* Config for dev */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};


/***/ })

}]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./defered.js: config-uat.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-uat\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for uat ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested, somethingElse */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingElse\\", function() { return somethingElse; });
/* Config for uat */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};
const somethingElse = \\"in uat\\";


/***/ })

}]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./defered.js: main.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// install a JSONP callback for chunk loading
/******/ 	function webpackJsonpCallback(data) {
/******/ 		var chunkIds = data[0];
/******/ 		var moreModules = data[1];
/******/
/******/
/******/ 		// add \\"moreModules\\" to the modules object,
/******/ 		// then flag all \\"chunkIds\\" as loaded and fire callback
/******/ 		var moduleId, chunkId, i = 0, resolves = [];
/******/ 		for(;i < chunkIds.length; i++) {
/******/ 			chunkId = chunkIds[i];
/******/ 			if(Object.prototype.hasOwnProperty.call(installedChunks, chunkId) && installedChunks[chunkId]) {
/******/ 				resolves.push(installedChunks[chunkId][0]);
/******/ 			}
/******/ 			installedChunks[chunkId] = 0;
/******/ 		}
/******/ 		for(moduleId in moreModules) {
/******/ 			if(Object.prototype.hasOwnProperty.call(moreModules, moduleId)) {
/******/ 				modules[moduleId] = moreModules[moduleId];
/******/ 			}
/******/ 		}
/******/ 		if(parentJsonpFunction) parentJsonpFunction(data);
/******/
/******/ 		while(resolves.length) {
/******/ 			resolves.shift()();
/******/ 		}
/******/
/******/ 	};
/******/
/******/
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// object to store loaded and loading chunks
/******/ 	// undefined = chunk not loaded, null = chunk preloaded/prefetched
/******/ 	// Promise = chunk loading, 0 = chunk loaded
/******/ 	var installedChunks = {
/******/ 		\\"main\\": 0
/******/ 	};
/******/
/******/
/******/
/******/ 	// script path function
/******/ 	function jsonpScriptSrc(chunkId) {
/******/ 		return __webpack_require__.p + \\"\\" + ({\\"config-dev\\":\\"config-dev\\",\\"config-uat\\":\\"config-uat\\"}[chunkId]||chunkId) + \\".js\\"
/******/ 	}
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/ 	// This file contains only the entry chunk.
/******/ 	// The chunk loading function for additional chunks
/******/ 	__webpack_require__.e = function requireEnsure(chunkId) {
/******/ 		var promises = [];
/******/
/******/
/******/ 		// JSONP chunk loading for javascript
/******/
/******/ 		var installedChunkData = installedChunks[chunkId];
/******/ 		if(installedChunkData !== 0) { // 0 means \\"already installed\\".
/******/
/******/ 			// a Promise means \\"currently loading\\".
/******/ 			if(installedChunkData) {
/******/ 				promises.push(installedChunkData[2]);
/******/ 			} else {
/******/ 				// setup Promise in chunk cache
/******/ 				var promise = new Promise(function(resolve, reject) {
/******/ 					installedChunkData = installedChunks[chunkId] = [resolve, reject];
/******/ 				});
/******/ 				promises.push(installedChunkData[2] = promise);
/******/
/******/ 				// start chunk loading
/******/ 				var script = document.createElement('script');
/******/ 				var onScriptComplete;
/******/
/******/ 				script.charset = 'utf-8';
/******/ 				script.timeout = 120;
/******/ 				if (__webpack_require__.nc) {
/******/ 					script.setAttribute(\\"nonce\\", __webpack_require__.nc);
/******/ 				}
/******/ 				script.src = jsonpScriptSrc(chunkId);
/******/
/******/ 				// create error before stack unwound to get useful stacktrace later
/******/ 				var error = new Error();
/******/ 				onScriptComplete = function (event) {
/******/ 					// avoid mem leaks in IE.
/******/ 					script.onerror = script.onload = null;
/******/ 					clearTimeout(timeout);
/******/ 					var chunk = installedChunks[chunkId];
/******/ 					if(chunk !== 0) {
/******/ 						if(chunk) {
/******/ 							var errorType = event && (event.type === 'load' ? 'missing' : event.type);
/******/ 							var realSrc = event && event.target && event.target.src;
/******/ 							error.message = 'Loading chunk ' + chunkId + ' failed.\\\\n(' + errorType + ': ' + realSrc + ')';
/******/ 							error.name = 'ChunkLoadError';
/******/ 							error.type = errorType;
/******/ 							error.request = realSrc;
/******/ 							chunk[1](error);
/******/ 						}
/******/ 						installedChunks[chunkId] = undefined;
/******/ 					}
/******/ 				};
/******/ 				var timeout = setTimeout(function(){
/******/ 					onScriptComplete({ type: 'timeout', target: script });
/******/ 				}, 120000);
/******/ 				script.onerror = script.onload = onScriptComplete;
/******/ 				document.head.appendChild(script);
/******/ 			}
/******/ 		}
/******/ 		return Promise.all(promises);
/******/ 	};
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"/\\";
/******/
/******/ 	// on error function for async loading
/******/ 	__webpack_require__.oe = function(err) { console.error(err); throw err; };
/******/
/******/ 	var jsonpArray = window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || [];
/******/ 	var oldJsonpFunction = jsonpArray.push.bind(jsonpArray);
/******/ 	jsonpArray.push = webpackJsonpCallback;
/******/ 	jsonpArray = jsonpArray.slice();
/******/ 	for(var i = 0; i < jsonpArray.length; i++) webpackJsonpCallback(jsonpArray[i]);
/******/ 	var parentJsonpFunction = oldJsonpFunction;
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./defered.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./defered.js\\":
/*!********************!*\\\\
  !*** ./defered.js ***!
  \\\\********************/
/*! no static exports found */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__.e(/*! import() */ 0).then(__webpack_require__.t.bind(null, /*! gdu/config */ 0, 7))
	.then(config => {
		console.log(config);
	});


/***/ })

/******/ });"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./index.js: config-dev.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-dev\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for dev ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* Config for dev */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};


/***/ })

}]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./index.js: config-uat.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-uat\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for uat ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested, somethingElse */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingElse\\", function() { return somethingElse; });
/* Config for uat */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};
const somethingElse = \\"in uat\\";


/***/ })

}]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot for ./index.js: main.js 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// install a JSONP callback for chunk loading
/******/ 	function webpackJsonpCallback(data) {
/******/ 		var chunkIds = data[0];
/******/ 		var moreModules = data[1];
/******/
/******/
/******/ 		// add \\"moreModules\\" to the modules object,
/******/ 		// then flag all \\"chunkIds\\" as loaded and fire callback
/******/ 		var moduleId, chunkId, i = 0, resolves = [];
/******/ 		for(;i < chunkIds.length; i++) {
/******/ 			chunkId = chunkIds[i];
/******/ 			if(Object.prototype.hasOwnProperty.call(installedChunks, chunkId) && installedChunks[chunkId]) {
/******/ 				resolves.push(installedChunks[chunkId][0]);
/******/ 			}
/******/ 			installedChunks[chunkId] = 0;
/******/ 		}
/******/ 		for(moduleId in moreModules) {
/******/ 			if(Object.prototype.hasOwnProperty.call(moreModules, moduleId)) {
/******/ 				modules[moduleId] = moreModules[moduleId];
/******/ 			}
/******/ 		}
/******/ 		if(parentJsonpFunction) parentJsonpFunction(data);
/******/
/******/ 		while(resolves.length) {
/******/ 			resolves.shift()();
/******/ 		}
/******/
/******/ 	};
/******/
/******/
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// object to store loaded and loading chunks
/******/ 	// undefined = chunk not loaded, null = chunk preloaded/prefetched
/******/ 	// Promise = chunk loading, 0 = chunk loaded
/******/ 	var installedChunks = {
/******/ 		\\"main\\": 0
/******/ 	};
/******/
/******/
/******/
/******/ 	// script path function
/******/ 	function jsonpScriptSrc(chunkId) {
/******/ 		return __webpack_require__.p + \\"\\" + ({\\"config-dev\\":\\"config-dev\\",\\"config-uat\\":\\"config-uat\\"}[chunkId]||chunkId) + \\".js\\"
/******/ 	}
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/ 	// This file contains only the entry chunk.
/******/ 	// The chunk loading function for additional chunks
/******/ 	__webpack_require__.e = function requireEnsure(chunkId) {
/******/ 		var promises = [];
/******/
/******/
/******/ 		// JSONP chunk loading for javascript
/******/
/******/ 		var installedChunkData = installedChunks[chunkId];
/******/ 		if(installedChunkData !== 0) { // 0 means \\"already installed\\".
/******/
/******/ 			// a Promise means \\"currently loading\\".
/******/ 			if(installedChunkData) {
/******/ 				promises.push(installedChunkData[2]);
/******/ 			} else {
/******/ 				// setup Promise in chunk cache
/******/ 				var promise = new Promise(function(resolve, reject) {
/******/ 					installedChunkData = installedChunks[chunkId] = [resolve, reject];
/******/ 				});
/******/ 				promises.push(installedChunkData[2] = promise);
/******/
/******/ 				// start chunk loading
/******/ 				var script = document.createElement('script');
/******/ 				var onScriptComplete;
/******/
/******/ 				script.charset = 'utf-8';
/******/ 				script.timeout = 120;
/******/ 				if (__webpack_require__.nc) {
/******/ 					script.setAttribute(\\"nonce\\", __webpack_require__.nc);
/******/ 				}
/******/ 				script.src = jsonpScriptSrc(chunkId);
/******/
/******/ 				// create error before stack unwound to get useful stacktrace later
/******/ 				var error = new Error();
/******/ 				onScriptComplete = function (event) {
/******/ 					// avoid mem leaks in IE.
/******/ 					script.onerror = script.onload = null;
/******/ 					clearTimeout(timeout);
/******/ 					var chunk = installedChunks[chunkId];
/******/ 					if(chunk !== 0) {
/******/ 						if(chunk) {
/******/ 							var errorType = event && (event.type === 'load' ? 'missing' : event.type);
/******/ 							var realSrc = event && event.target && event.target.src;
/******/ 							error.message = 'Loading chunk ' + chunkId + ' failed.\\\\n(' + errorType + ': ' + realSrc + ')';
/******/ 							error.name = 'ChunkLoadError';
/******/ 							error.type = errorType;
/******/ 							error.request = realSrc;
/******/ 							chunk[1](error);
/******/ 						}
/******/ 						installedChunks[chunkId] = undefined;
/******/ 					}
/******/ 				};
/******/ 				var timeout = setTimeout(function(){
/******/ 					onScriptComplete({ type: 'timeout', target: script });
/******/ 				}, 120000);
/******/ 				script.onerror = script.onload = onScriptComplete;
/******/ 				document.head.appendChild(script);
/******/ 			}
/******/ 		}
/******/ 		return Promise.all(promises);
/******/ 	};
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"/\\";
/******/
/******/ 	// on error function for async loading
/******/ 	__webpack_require__.oe = function(err) { console.error(err); throw err; };
/******/
/******/ 	var jsonpArray = window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || [];
/******/ 	var oldJsonpFunction = jsonpArray.push.bind(jsonpArray);
/******/ 	jsonpArray.push = webpackJsonpCallback;
/******/ 	jsonpArray = jsonpArray.slice();
/******/ 	for(var i = 0; i < jsonpArray.length; i++) webpackJsonpCallback(jsonpArray[i]);
/******/ 	var parentJsonpFunction = oldJsonpFunction;
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./index.js\\":
/*!******************!*\\\\
  !*** ./index.js ***!
  \\\\******************/
/*! no exports provided */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var gdu_config__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! gdu/config */ 0);
/* harmony import */ var gdu_config__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(gdu_config__WEBPACK_IMPORTED_MODULE_0__);



console.log(
	{
		config: gdu_config__WEBPACK_IMPORTED_MODULE_0__,
		somethingValue: gdu_config__WEBPACK_IMPORTED_MODULE_0__[\\"somethingValue\\"],
		nestedRenamedToSomethingElse: gdu_config__WEBPACK_IMPORTED_MODULE_0__[\\"nested\\"],
	},
);


/***/ }),

/***/ 0:
/*!********************!*\\\\
  !*** config query ***!
  \\\\********************/
/*! no static exports found */
/*! all exports used */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony import */ var __CONFIG__ = __webpack_require__(/*! config_module */ \\"config_module\\");
var __CONFIG___namespace = /*#__PURE__*/__webpack_require__.t(/*! config_module */ \\"config_module\\", 1);
/* harmony reexport (unknown) */ for(var __WEBPACK_IMPORT_KEY__ in __CONFIG__) if(__WEBPACK_IMPORT_KEY__ !== 'default') (function(key) { __webpack_require__.d(exports, key, function() { return __CONFIG__[key]; }) }(__WEBPACK_IMPORT_KEY__));


/***/ })

/******/ });"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot: config-dev.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-dev\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for dev ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* Config for dev */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};


/***/ })

}]);"
`;

exports[`ConfigsWebpackPlugin :: RuntimeConfigsPlugin :: basic should match snapshot: config-uat.js 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[\\"config-uat\\"],{

/***/ \\"config_module\\":
/*!**********************!*\\\\
  !*** config for uat ***!
  \\\\**********************/
/*! exports provided: somethingValue, shouldTreeShake, nested, somethingElse */
/***/ (function(module, exports, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(exports);
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingValue\\", function() { return somethingValue; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"shouldTreeShake\\", function() { return shouldTreeShake; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"nested\\", function() { return nested; });
/* harmony export (binding) */ __webpack_require__.d(exports, \\"somethingElse\\", function() { return somethingElse; });
/* Config for uat */

const somethingValue = \\"fooBar\\";
const shouldTreeShake = \\"tree-shaken-value\\";
const nested = {
    \\"test\\": \\"nested\\"
};
const somethingElse = \\"in uat\\";


/***/ })

}]);"
`;
